numpyro/distributions/distribution.py

```diff
@@ -127,12 +128,13 @@ def logpdf(self, x, *args, **kwargs):
             return super(jax_continuous, self).logpdf(x, *args, **kwargs)
 
 
-class jax_discrete(osp_stats.rv_discrete):
-    args_check = True
-
+class jax_discrete(jax_generic, osp_stats.rv_discrete):
     def _support_mask(self, k):
         return (k >= self.a) & (k <= self.b) & (np.floor(k) == k)
 
+    def _support(self, *args, **kwargs):
+        return self._support_mask
```

Comment:
Though it is not important because we don't want to work with `loc!=0`, could you add a TODO comment here to remind us in case this scipy interface works well on its own?